File: javascript-nodejs/receive_logs.js
var amqp = require('amqp');

var connection = amqp.createConnection({host: 'localhost'});

connection.on('ready', function(){
    connection.exchange('logs', {type: 'fanout',
                                 autoDelete: false}, function(exchange){
        connection.queue('tmp-' + Math.random(), {exclusive: true},
                         function(queue){
            queue.bind('logs', '');
            console.log(' [*] Waiting for logs. To exit press CTRL+C')

            queue.subscribe(function(msg){
                console.log(" [x] %s", msg.data.toString('utf-8'));
            });
        })
    });
});

File: java-amqp/src/main/java/ReceiveLogs.java
import com.rabbitmq.client.amqp.Connection;
import com.rabbitmq.client.amqp.Consumer;
import com.rabbitmq.client.amqp.Environment;
import com.rabbitmq.client.amqp.Management;
import com.rabbitmq.client.amqp.Management.QueueInfo;

import java.nio.charset.StandardCharsets;

public class ReceiveLogs {

    private static final String EXCHANGE_NAME = "logs";

    public static void main(String[] argv) throws Exception {
        Environment environment = TutorialSupport.newEnvironment();
        Connection connection = environment.connectionBuilder().uri(TutorialSupport.BROKER_URI).build();

        String queueName;
        try (Management management = connection.management()) {
            management.exchange().name(EXCHANGE_NAME).type(Management.ExchangeType.FANOUT).declare();
            QueueInfo q = management.queue().exclusive(true).autoDelete(true).declare();
            queueName = q.name();
            management.binding()
                    .sourceExchange(EXCHANGE_NAME)
                    .destinationQueue(queueName)
                    .key("")
                    .bind();
        }

        System.out.println(" [*] Waiting for messages. To exit press CTRL+C");

        Consumer consumer = connection.consumerBuilder()
                .queue(queueName)
                .messageHandler((ctx, message) -> {
                    String body = new String(message.body(), StandardCharsets.UTF_8);
                    System.out.println(" [x] Received '" + body + "'");
                    ctx.accept();
                })
                .build();

        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
            consumer.close();
            connection.close();
            environment.close();
        }));

        Thread.sleep(Long.MAX_VALUE);
    }
}
